src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -1533,6 +1550,8 @@ internal partial class BoundQueryClause
     {
         protected override OperationKind ExpressionKind => OperationKind.None;
 
+        protected override ImmutableArray<IOperation> Children => ImmutableArray.Create<IOperation>(this.Value);
```

Comment:
How do we ensure that we make callbacks for any operation nodes for user code within query? I think the fact that we are returning compiler generated nodes is irrelevant to this approach or the IOperationWithChildren API.